codemp/rd-rend2/glsl/lightall.glsl

```diff
@@ -499,6 +500,50 @@ float CalcLightAttenuation(float point, float normDist)
 	return clamp(attenuation, 0.0, 1.0);
 }
 
+#if defined(USE_LIGHT_VECTOR) && !defined(USE_VERTEX_LIGHTING) && defined(USE_DSHADOWS)
+#define DEPTH_MAX_ERROR 0.000000059604644775390625
+
+vec3 sampleOffsetDirections[20] = vec3[]
+(
+	vec3(1, 1, 1), vec3(1, -1, 1), vec3(-1, -1, 1), vec3(-1, 1, 1),
```

Comment:
1. Make these explicitly floats (`1.0`, `0.0`, etc)
2. 20 seems excessive?
3. Can probably make use of hardware PCF too by specifying half offsets. That way the GPU will do PCF filtering on 4 samples at a time (basically free)

Reply:
20 seems to be pretty fast this way. I could add r_dlightMode 3 for this pcf filtering and make r_dlightmode 2 use the buildin 2x2 pcf. It should do 4 samples by default with a shadowSampler afaik.